amazon-kinesis-client/src/test/java/software/amazon/kinesis/application/TestRecordProcessorFactory.java

```diff
@@ -14,7 +15,12 @@ public TestRecordProcessorFactory(RecordValidatorQueue recordValidator) {
 
     @Override
     public ShardRecordProcessor shardRecordProcessor() {
-        return new TestRecordProcessor(this.recordValidator);
+        return new TestRecordProcessor(null, this.recordValidator);
```

Comment:
do we ever expect this method to be called now that we're implementing `ShardRecordProcessorFactory#shardRecordProcessor(StreamIdentifier)`?
if not, should we just throw instead?